### ISSUE TITLE
Compact Multi-Class Boosted Trees

### ISSUE THREAD

**Issue by samanemami** [tensorflow/tensorflow#48907]
<em>The issue related to the performance of the [BoostedTree](https://github.com/tensorflow/estimator/blob/781c0d30c6bf100aa174591dd97cb70fc39d294d/tensorflow_estimator/python/estimator/canned/boosted_trees_test.py#L403) classifier in terms of accuracy.</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):
- OS Platform and Distribution: Linux Ubuntu 16.04, Windows 10
- TensorFlow installed from (source or binary): pip install TensorFlow 
- TensorFlow version (use command below): 2.4.1
- Python version: 3.6

**Describe the model**
BoostedTree classifier is a model introduced by [N Ponomareva, T Colthurst, G Hendry.2017](https://ieeexplore.ieee.org/abstract/document/8257910).
It is built over the Xgboost idea and learning is done through building one layer of decision tree regressor over N boosting iteration.

**Describe the expected behavior**
As the related paper discovered, for ten trees, the accuracy of the xgboost is about 60%, and for the TF model is 73%.
This experiment had done over the Cover type dataset.
**Describe the current behavior**
I did the same [experiments](https://github.com/samanemami/TFBoostedTree/blob/main/examples/Comparison.ipynb) over the same dataset and identical configuration, but the results have a huge difference.
In this experiment, the accuracy of xgboost is about 70%, and the TF model score is about 58 percent.
It is worthy to note the fact the logistic regression at least achieves 70 percent.

About the training time, if you plot the training time over boosting iteration you will have a Neutral trend which should be linear as the gradient boosting is a linear ensemble mode.

**Standalone code to reproduce the issue**
I reproduced the model [here](https://github.com/samanemami/TFBoostedTree/blob/main/TFBT.py).
Also, you may find the related experiments [here](https://github.com/samanemami/TFBoostedTree/tree/main/examples).

**Comment by saikumarchalla** [comment]
@samanemami  I tried to reproduce the issue  but facing error while importing data. Please check the gist [here](https://colab.research.google.com/gist/saikumarchalla/848e49f1a2d1e1799c88c8183496599c/comparison.ipynb).

**Comment by samanemami** [comment]
@saikumarchalla Please find the modified gist [here](https://colab.research.google.com/gist/samanemami/b1a4797ad00e6c84c61a9a00fc363127/comparison.ipynb)

**Comment by saikumarchalla** [comment]
Was able to reproduce the issue using TF version [2.4](https://colab.research.google.com/gist/saikumarchalla/449304216e9f97f6a14ecf22df52ef9f/comparison.ipynb#scrollTo=advanced-andrews) and [Nightly](https://colab.research.google.com/gist/saikumarchalla/305dda77526c4903f5b612105ee0466d/comparison.ipynb) version. Thanks!

**Comment by samanemami** [comment]
After running different experiments, I found that you reduced the tree number in XGBoost, hence, it reduced the XGBoost accuracy.
